app/src/main/java/com/example/universitieslist/util/HandleApiCall.kt

```diff
@@ -2,29 +2,26 @@ package com.example.universitieslist.util
 
 import com.example.universitieslist.util.Constants.UNKNOWN_ERROR
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.flow
 import retrofit2.Response
 import timber.log.Timber
 
 suspend fun <T> performApiCall(
     apiCall: suspend () -> Response<T>,
     country: String,
-): Flow<T> = flow {
+): Flow<Result<T?>> = flow {
```

Comment:
You don't necessarily need to use `Flow` here. Passing the result in the wrapper would suffice. The `Flow` should be used in the function making the network call.

Reply:
I tried removing it but seems it's needed because of the `emit()` function used to emit success and failure results.